paasta_tools/kubernetes_tools.py

```diff
@@ -3124,6 +3144,27 @@ def create_or_find_service_account_name(
             ),
         )
         kube_client.core.create_namespaced_service_account(namespace=namespace, body=sa)
+
+        # we're expecting that any Role dynamically associated with a Service Account already exist.
+        # at Yelp, this means that we have a version-controlled resource for the Role in Puppet.
+        if k8s_role:
+            # since the Role already exists, we just need to associate it with the Service Account through
+            # a Role Binding
+            role_binding = V1RoleBinding(
+                metadata=V1ObjectMeta(name=sa_name, namespace=namespace,),
+                role_ref=V1RoleRef(
+                    api_group="rbac.authorization.k8s.io", kind="Role", name=k8s_role,
+                ),
+                subjects=[
+                    V1Subject(
+                        kind="ServiceAccount", namespace=namespace, name=sa_name,
+                    ),
+                ],
+            )
+            kube_client.rbac.create_namespaced_role_binding(
+                namespace=namespace, body=role_binding
+            )
+
```

Comment:
not a blocker, just thinking out loud -- Since this role binding is only created where we're also creating the service account, do you think there's anything we should do here to keep a close eye on whether we somehow create the service account successfully but fail to create the role binding? 

It seems like we could get into a place where the rolebinding is missing and the setup_tron_namespace job would never complain about it.

If a tronjob starts yelling about privs, the manual workaround/fix would be to delete the service account which seems pretty manageable to start though.

Reply:
good catch - lemme refactor this a bit so that we always check that both the Service Account and the RoleBinding exist

might as well have computers deal with this rather than have words in a runbook that you have to know exist :p